pkgs/applications/networking/charles/default.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv, fetchurl, makeDesktopItem, jre, hicolor-icon-theme }:
+
+let
+  desktopItem = makeDesktopItem {
+    categories = "Network;Development;WebDevelopment;Java;";
+    desktopName = "Charles";
+    exec = "charles %F";
+    genericName  = "Web Debugging Proxy";
+    icon = "charles-proxy";
+    mimeType = "application/x-charles-savedsession;application/x-charles-savedsession+xml;application/x-charles-savedsession+json;application/har+json;application/vnd.tcpdump.pcap;application/x-charles-trace";
+    name = "Charles";
+    startupNotify = "true";
+  };
+
+in stdenv.mkDerivation rec {
+  name = "charles-${version}";
+  version = "4.2.6";
+
+  src = fetchurl {
+    url = "https://www.charlesproxy.com/assets/release/${version}/charles-proxy-${version}.tar.gz";
+    sha256 = "1hjfimyr9nnbbxadwni02d2xl64ybarh42l1g6hlslq5qwl8ywzb";
+  };
+
+  installPhase = ''
+    mkdir -pv $out/bin
+
+    cat > $out/bin/charles << EOF
```

Comment:
it could be done with ```makeWrapper```

[Example](https://github.com/NixOS/nixpkgs/blob/5b9e29c01cafd6054b5f4caae4615bef14e6e4ef/pkgs/development/tools/nailgun/default.nix#L29)

Reply:
done